### PR TITLE
Add typescript definitions

### PR DESCRIPTION
**Pull request by sangdth**
Add typescript definitions file index.d.ts, also there were some changes to eslint config to make it work with typescript keywords.

### CODE REVIEW

**Comment by sangdth** [comment]
I realized that the file `tsconfig.json` is useless. I will delete it in next commit. Just wondering, is there anything else I need to do as well? Also I could pump up the version, couldn't I?

Thanks.

**Comment by doowb** [comment]
Thanks! I'll review this shortly.

You don't need to bump the version, that's better done after the code is merged in.

**Comment by jaruszewskiG** [comment]
Bump, waiting for this PR to be merged.

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Type definitions for randomatic 3.1
+// Project: https://github.com/jonschlinkert/randomatic
+// Definitions by: Sang Dang <https://github.com/sangdth>
+
+interface randomOptions {
+  chars: string;
```
Comment (lukaselmer) [inline index.d.ts]: Shouldn't `chars` be `chars?`?

Comment (sangdth) [reply index.d.ts:6]: In typescript, there is only `string`

Comment (lukaselmer) [reply index.d.ts:6]: `chars?: string;` instead of `chars: string;` 😉

Comment (lukaselmer) [reply index.d.ts:6]: image

https://github.com/jonschlinkert/randomatic#options

Comment (lukaselmer) [reply index.d.ts:6]: https://github.com/jonschlinkert/randomatic/blob/master/test.js#L107

Comment (sangdth) [reply index.d.ts:6]: Ha ha I got it. How can I miss the second question mark :)) I will fix it.

**Pull request merged**